Assets/Mirror/Runtime/SyncDictionary.cs

```diff
@@ -0,0 +1,331 @@
+                        }
+                        break;
+
+                    case Operation.OP_SET:
+                    case Operation.OP_DIRTY:
+                        key = DeserializeKey(reader);
+                        item = DeserializeItem(reader);
+                        if (apply)
+                        {
+                            m_Objects[key] = item;
+                        }
+                        break;
+                }
+
+                if (apply)
+                {
+                    Callback?.Invoke(operation, key, item);
+                }
+                // we just skipped this change
+                else
+                {
+                    changesAhead--;
+                }
+            }
+        }
+
+
+        public void Clear()
+        {
+            m_Objects.Clear();
+            AddOperation(Operation.OP_CLEAR, default);
+        }
+
+        public bool ContainsKey(B key) => m_Objects.ContainsKey(key);
+
+        public bool Remove(B key)
+        {
+            bool result = m_Objects.Remove(key);
+            if (result)
+            {
+                AddOperation(Operation.OP_REMOVE, key);
+            }
+            return result;
+        }
+
+        public void Dirty(B index)
+        {
+            AddOperation(Operation.OP_DIRTY, index, m_Objects[index]);
+        }
+
+        public T this[B i]
+        {
+            get { return m_Objects[i]; }
+            set
+            {
+                bool changed = false;
+                if (m_Objects[i] == null)
+                {
+                    if (value == null)
+                        return;
+                    else
+                        changed = true;
+                }
+                else
+                {
+                    changed = !m_Objects[i].Equals(value);
+                }
+
+                m_Objects[i] = value;
+                if (changed)
+                {
+                    AddOperation(Operation.OP_SET, i, value);
+                }
+            }
+        }
+
+        public bool TryGetValue(B key, out T value)
+        {
+            if (m_Objects.ContainsKey(key))
+            {
+                value = m_Objects[key];
+                return true;
+            }
+            else
+            {
+                value = default;
+                return false;
+            }
+        }
+
+        public void Add(B key, T value)
+        {
+            m_Objects.Add(key, value);
+            AddOperation(Operation.OP_ADD, key, value);
+        }
+
+        public void Add(KeyValuePair<B, T> item)
+        {
+            m_Objects.Add(item.Key, item.Value);
+            AddOperation(Operation.OP_ADD, item.Key, item.Value);
+        }
+
+        public bool Contains(KeyValuePair<B, T> item)
+        {
+            return m_Objects.ContainsKey(item.Key) && m_Objects.ContainsValue(item.Value);
+        }
+
+        public void CopyTo(KeyValuePair<B, T>[] array, int arrayIndex)
```

Comment:
Why do we need this method?  it is not part of IDictionary

Reply:
It is, unfortunately, as IDictionary implements ICollection.

Reply:
ok,  then can you just call the m_Objects.CopyTo instead?

Reply:
Dictionary does not implement CopyTo sadly.